README.md

```diff
@@ -63,6 +63,13 @@ class ObligatoryCryptocurrencyReference extends Component {
 }
 ```
 
+```js
+const responses = await AsyncIterator.from(urls).map(async (url) => {
```

Comment:
I think this case doesn't correct.

The `AsyncIterator.from(urls).map()` part is converting a `string[]` to `AsyncIterator<string>` and then map it to a `AsyncIterator<JSONValue>`. The `AsyncIterator` itself is not an `then`able object so `responses` are still `AsyncIterator<JSONValue>`. You need to use the `for await ... of` syntax to consume it.

Reply:
Oh no, so I guess [the example](https://github.com/tc39/proposal-async-iteration/issues/134#issuecomment-711422945) from @Jamesernator is also incorrect?

Reply:
Wonder if there's still a way to get this type of syntax sugar without `for await ... of`.

Really like the functional look of this pattern of being able to map over an array with an async function as the mapping function, like `p-map` does: https://github.com/sindresorhus/p-map

Reply:
You can chain `.toArray()` at the end I guess, it should turn `AsyncIterator<T>` into `Promise<T[]>. But if you do so, there is no actually difference between this way:

```js
await Promise.all(urls.map(async URL => { ... }))
```

Reply:
use .collect at the end to create an array.

Reply:
Oh, but wouldn't `Promise.all` run them all at once, and using `.map` run them sequentially?

Reply:
Promise.all doesn't "run" anything - it just waits for the results of work you've already started. The `.map` callback is what "runs" a task.

Reply:
> Oh, but wouldn't `Promise.all` run them all at once, and using `.map` run them sequentially?
> 
> That's the main reason for avoiding `Promise.all` right now.

You're right, my example is parallel and your example is serial.

Reply:
`await urls.reduce(async (prev, URL) => { await prev; ... }, Promise.resolve())` is the serial loopless version.

Reply:
Nice yeah, I've used that `reduce` pattern before!

It's nice, but I was hoping that there was a serial `.map` with async functions in this proposal somewhere....

Reply:
> Promise.all doesn't "run" anything

Yeah, I probably wasn't being clear in the way I spoke. I meant "the `Promise.all` version"